modules/msk-serverless/main.tf

```diff
@@ -0,0 +1,250 @@
+        content {
+          scram = var.enable_client_sasl_scram
+          iam   = var.enable_client_sasl_iam
+        }
+      }
+    }
+  }
+
+  logging_info {
+    broker_logs {
+      cloudwatch_logs {
+        enabled   = var.enable_cloudwatch_logs
+        log_group = var.cloudwatch_log_group
+      }
+      firehose {
+        enabled         = var.enable_firehose_logs
+        delivery_stream = var.firehose_delivery_stream
+      }
+      s3 {
+        enabled = var.enable_s3_logs
+        bucket  = var.s3_logs_bucket
+        prefix  = var.s3_logs_prefix
+      }
+    }
+  }
+
+  open_monitoring {
+    prometheus {
+      jmx_exporter {
+        enabled_in_broker = var.open_monitoring_enable_jmx_exporter
+      }
+      node_exporter {
+        enabled_in_broker = var.open_monitoring_enable_node_exporter
+      }
+    }
+  }
+
+  tags = var.custom_tags
+
+  # Ignore changes to the broker volume, because 'aws_appautoscaling_policy' will take care of scaling the volume
+  lifecycle {
+    ignore_changes = [broker_node_group_info[0].ebs_volume_size]
+  }
+}
+
+# AWS Secrets Manager holds entries for username and password authentication for a cluster.
+# You will have to create the Secrets Manager secrets separately. For full details, see
+# https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/msk_scram_secret_association
+resource "aws_msk_scram_secret_association" "default" {
+  count = var.enable_client_sasl_scram && length(var.client_sasl_scram_secret_arns) > 0 ? 1 : 0
+
+  cluster_arn     = aws_msk_cluster.msk.arn
+  secret_arn_list = var.client_sasl_scram_secret_arns
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# BROKER STORAGE AUTOSCALING
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "aws_appautoscaling_target" "msk" {
+  max_capacity       = var.broker_storage_autoscaling_max_capacity
+  min_capacity       = 1
+  role_arn           = var.override_broker_storage_autoscaling_role_arn
+  resource_id        = aws_msk_cluster.msk.arn
+  scalable_dimension = "kafka:broker-storage:VolumeSize"
+  service_namespace  = "kafka"
+}
+
+resource "aws_appautoscaling_policy" "msk" {
+  name               = "${var.cluster_name}-broker-storage-scaling"
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = aws_msk_cluster.msk.arn
+  scalable_dimension = aws_appautoscaling_target.msk.scalable_dimension
+  service_namespace  = aws_appautoscaling_target.msk.service_namespace
+  target_tracking_scaling_policy_configuration {
+    disable_scale_in = var.disable_broker_storage_scale_in
+    predefined_metric_specification {
+      predefined_metric_type = "KafkaBrokerStorageUtilization"
+    }
+
+    target_value = var.broker_storage_autoscaling_target_percentage
+  }
+}
```

Comment:
Once we've confirmed what options we want to use, this module + variables/outputs can be simplified.